### PR TITLE
Switch history

### LINKED ISSUE #349: Method to open a history in the browser
Maybe it's just me with this use-case, but I find myself switching often between bioblend and the UI when I'm working. When dealing with thousands of histories it can be a bit of a pain to find the history in the UI which I was just working on in bioblend. It would be handy to have a `gi.histories.switch_to(history_id)` which I can just run, go to the browser window and refresh.

This needs a change to the Galaxy API but it's still simple enough to count as a paper cut.

### PR DESCRIPTION
**Pull request by simonbray**
Fixes #349.

### CODE REVIEW

File: bioblend/galaxy/histories/__init__.py [hunk 1]
```diff
@@ -687,3 +688,19 @@ def copy_dataset(self, history_id, dataset_id, source='hda'):
...
+        corresponding to the history ID provided.
+
+        :type history_id: str
+        :param history_id: history ID to switch to
+
+        :rtype: None
+        :return: `None`
+        """
+
+        url = ''.join((self.gi.base_url.strip('/'), '/history/switch_to_history?hist_id=', history_id))
```
Comment (nsoranzo) [inline bioblend/galaxy/histories/__init__.py:704]:
```suggestion
        url = urljoin(self.gi.base_url, f"history/switch_to_history?hist_id={history_id}")
```

File: bioblend/galaxy/histories/__init__.py [hunk 2]
```diff
@@ -687,3 +688,19 @@ def copy_dataset(self, history_id, dataset_id, source='hda'):
 
         url = self._make_url(history_id, contents=True)
         return self._post(payload=payload, url=url)
+
+    def switch_to_history(self, history_id):
```
Comment (nsoranzo) [inline bioblend/galaxy/histories/__init__.py:692]:
```suggestion
    def open_history(self, history_id):
```

File: bioblend/galaxy/histories/__init__.py [hunk 3]
```diff
@@ -687,3 +688,19 @@ def copy_dataset(self, history_id, dataset_id, source='hda'):
...
+        """
+        Opens a new tab in the default browser and switches to the history
+        corresponding to the history ID provided.
```
Comment (nsoranzo) [inline bioblend/galaxy/histories/__init__.py:695]:
```suggestion
        Open Galaxy in a new tab of the default web browser and switch to the
        specified history.
```

File: bioblend/galaxy/histories/__init__.py [hunk 4]
```diff
@@ -687,3 +688,19 @@ def copy_dataset(self, history_id, dataset_id, source='hda'):
...
+
+        :type history_id: str
+        :param history_id: history ID to switch to
+
+        :rtype: None
+        :return: `None`
+        """
```
Comment (nsoranzo) [inline bioblend/galaxy/histories/__init__.py:702]:
```suggestion

        .. warning::
          After opening the specified history, all previously opened Galaxy tabs
          in the browser session will have the current history changed to this
          one, even if the interface still shows another history. Refreshing
          any such tab is recommended.
        """
```

File: bioblend/galaxy/histories/__init__.py (hunk 1 above)
Comment (simonbray) [reply bioblend/galaxy/histories/__init__.py:704]: Good to know, thanks

File: bioblend/galaxy/histories/__init__.py [hunk 5]
```diff
@@ -6,6 +6,7 @@
 import re
 import sys
 import time
+import webbrowser
```
Comment (nsoranzo) [inline bioblend/galaxy/histories/__init__.py:9]:
```suggestion
import webbrowser
from urllib.parse import urljoin
```

File: bioblend/galaxy/histories/__init__.py [hunk 6]
```diff
@@ -687,3 +689,24 @@ def copy_dataset(self, history_id, dataset_id, source='hda'):
...
+    def open_history(self, history_id):
+        """
+        Open Galaxy in a new tab of the default web browser and switch to the
+        specified history.
+
+        :type history_id: str
+        :param history_id: ID of the history to switch to
+
+        :rtype: None
+        :return: `None`
```
Comment (nsoranzo) [inline bioblend/galaxy/histories/__init__.py:702]:
```suggestion
        :rtype: NoneType
        :return: ``None``
```

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**